crossdock/client/tchserver/raw.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tchserver
+
+import (
+	"net"
+	"time"
+
+	"github.com/yarpc/yarpc-go"
+	"github.com/yarpc/yarpc-go/crossdock/client/behavior"
+	"github.com/yarpc/yarpc-go/crossdock/client/random"
+	"github.com/yarpc/yarpc-go/encoding/raw"
+	"github.com/yarpc/yarpc-go/transport"
+	"golang.org/x/net/context"
+)
+
+func runRaw(s behavior.Sink, rpc yarpc.RPC) {
+	assert := behavior.Assert(s)
+	checks := behavior.Checks(s)
+
+	// TODO headers should be at yarpc, not transport
+	headers := transport.Headers{
+		"hello": "raw",
+	}
+	token := random.Bytes(5)
+
+	resBody, resMeta, err := rawCall(rpc, headers, token)
+
+	switch err.(type) {
+	case *net.OpError:
+		behavior.Skipf(s, "no tchannel server available: %v", err)
+		return
```

Comment:
Consider creating a function `connectionWasRefused(error) bool` so that we can make the check more sophisticated later because `OpError` can represents errors besides connection refused.